### ISSUE TITLE
Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by skbergam** [typescript-eslint/typescript-eslint#1608]
**Repro**

`package.json`:
```JSON
{
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^2.19.2",
    "@typescript-eslint/parser": "^2.19.2",
    "eslint": "^6.8.0",
    "eslint-config-airbnb": "^18.0.1",
    "eslint-config-prettier": "^6.10.0",
    "eslint-import-resolver-typescript": "^2.0.0",
    "eslint-plugin-import": "^2.20.1",
    "eslint-plugin-json": "^2.0.1",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-prettier": "^3.1.2",
    "eslint-plugin-react": "^7.18.3",
    "prettier": "^1.19.1",
    "typescript": "^3.7.5"
  }
}
```
`.eslintrc.js`:
```js
module.exports = {
  extends: ['airbnb', 'plugin:@typescript-eslint/recommended'],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint', 'prettier'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {},
    },
  },
  rules: {
    'react/jsx-filename-extension': [2, { extensions: ['.js', '.jsx', '.ts', '.tsx'] }],
    'import/no-extraneous-dependencies': [2, { devDependencies: ['**/test.tsx', '**/test.ts'] }],
    '@typescript-eslint/indent': [2, 2],
  },
};
```

`index.ts`:
```TS
console.log("FF");
```

**Expected Result**
Either no errors or at least some actual linter output.

**Actual Result**
```
Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot convert undefined or null to object
Referenced from: /Users/sean/git/ts-eslint-repro/.eslintrc.js
```

**Additional Info**

```
$ npx eslint . --debug --ext .js,.jsx,.ts,.tsx
  eslint:cli CLI args: [ '.', '--debug', '--ext', '.js,.jsx,.ts,.tsx' ] +0ms
  eslint:cli Running on files +4ms
  eslint:config-array-factory Loading .eslintignore file: /Users/sean/git/ts-eslint-repro/.eslintignore +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/sean/git/ts-eslint-repro', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/sean/git/ts-eslint-repro', patterns: [ '/node_modules/*', '/bower_components/*' ] } +2ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/sean/git/ts-eslint-repro', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/sean/git/ts-eslint-repro', patterns: [ '/node_modules/*', '/bower_components/*' ] } +1ms
  eslint:file-enumerator Start to iterate files: [ '.' ] +0ms
  eslint:file-enumerator Directory: /Users/sean/git/ts-eslint-repro +0ms
  eslint:file-enumerator Enter the directory: /Users/sean/git/ts-eslint-repro +1ms
  eslint:cascading-config-array-factory Load config files for /Users/sean/git/ts-eslint-repro. +0ms
  eslint:cascading-config-array-factory No cache found: /Users/sean/git/ts-eslint-repro. +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/.eslintrc.js +6ms
  eslint:config-array-factory Config file found: /Users/sean/git/ts-eslint-repro/.eslintrc.js +1ms
  eslint:config-array-factory Loading {extends:"airbnb"} relative to /Users/sean/git/ts-eslint-repro/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-config-airbnb@18.0.1 (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js) +1ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/best-practices.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +4ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/best-practices.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/best-practices.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/best-practices.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/best-practices.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/errors.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/errors.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/errors.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/errors.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/errors.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/node.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/node.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/node.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/node.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/node.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/style.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/style.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/style.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/style.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/style.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/variables.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +2ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/variables.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/variables.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/variables.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/variables.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/es6.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/es6.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/es6.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/es6.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/es6.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js +0ms
  eslint:config-array-factory Loading plugin "import" from /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/imports.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.20.1 (/Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-import/lib/index.js) +0ms
  eslint:config-array-factory Plugin /Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-import/lib/index.js loaded in: 89ms +89ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/strict.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/strict.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/strict.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/strict.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb-base/rules/strict.js +0ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +0ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js +0ms
  eslint:config-array-factory Loading plugin "react" from /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react.js +3ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.18.3 (/Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-react/index.js) +1ms
  eslint:config-array-factory Plugin /Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-react/index.js loaded in: 98ms +98ms
  eslint:config-array-factory Loading {extends:"/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js"} relative to /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module '/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js/package.json'
Require stack:
- /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/index.js +1ms
  eslint:config-array-factory Loaded: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js (/Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js) +0ms
  eslint:config-array-factory Loading JS config file: /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js +0ms
  eslint:config-array-factory Loading plugin "jsx-a11y" from /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-jsx-a11y@6.2.3 (/Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-jsx-a11y/lib/index.js) +2ms
  eslint:config-array-factory Plugin /Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-jsx-a11y/lib/index.js loaded in: 153ms +153ms
  eslint:config-array-factory Loading plugin "react" from /Users/sean/git/ts-eslint-repro/node_modules/eslint-config-airbnb/rules/react-a11y.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.18.3 (/Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-react/index.js) +1ms
  eslint:config-array-factory Plugin /Users/sean/git/ts-eslint-repro/node_modules/eslint-plugin-react/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/recommended"} relative to /Users/sean/git/ts-eslint-repro/.eslintrc.js +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/sean/git/ts-eslint-repro/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.19.2 (/Users/sean/git/ts-eslint-repro/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js'. +221ms
Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot convert undefined or null to object
Referenced from: /Users/sean/git/ts-eslint-repro/.eslintrc.js
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.2` |
| `@typescript-eslint/parser`        | `2.19.2` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `8.6.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
Could you provide a repro?
We don't have any other reports of this and it's working fine for me.

**Comment by skbergam** [comment]
Hi @bradzacher. Do you mean put the code up in a repo somewhere?

**Comment by bradzacher** [comment]
Yup. Please create a small, self contained set of code which consistently reproduces the issue, push it to a new github repo, and link it here.

**Comment by nathanielcook** [comment]
I ran into this problem with node 8.9.4 and fixed it by switching to node 12.16.1.

**Comment by bradzacher** [comment]
Oh, thanks, I didn't even notice the listed node version.
This is definitely intended not to be working.
Our tooling requires a minimum of node `8.10.0`, as listed in the package.json:

https://github.com/typescript-eslint/typescript-eslint/blob/9edd863b2a66ee44bd4a439903973e6c207480aa/packages/eslint-plugin/package.json#L12
